server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/view/widget/RecordPanel.java

```diff
@@ -162,9 +162,14 @@ public void reset() {
     }
     
     public boolean validate() {
-        return recordFieldWidget.validate();
+        return recordFieldWidget.validate() && isEmpty(recordFieldWidget);
     }
-    
+
+    private boolean isEmpty(RecordFieldWidget recordFieldWidget) {
+        return !(recordFieldWidget.getValue() != null && !recordFieldWidget.getValue().getValue().isEmpty());
+    }
+
+
     public void setFormDataLoader(FormDataLoader formDataLoader) {
```

Comment:
Don't use isEmpty in validate method. isEmpty should be separate public method. Also recordFieldWidget is already member of RecordPanel class, just access it directly without passing it as argument.